src/lib/comms/include/sol-bluetooth.h

```diff
@@ -0,0 +1,327 @@
+ *
+ * @param conn The reference to a connection.
+ *
+ * @return The same connection, with refcount increased, or NULL if invalid.
+ */
+struct sol_bt_conn *sol_bt_conn_ref(struct sol_bt_conn *conn);
+
+/**
+ * @brief Decreases the reference count of a connection.
+ *
+ * When the last reference is released, the connection and all the resources
+ * associated with it are released.
+ *
+ * @param conn The reference to a connection.
+ */
+void sol_bt_conn_unref(struct sol_bt_conn *conn);
+
+/**
+ * @brief Returns the network address of the remote device.
+ *
+ * @param conn The reference to a connection.
+ *
+ * @return The network link address associated with the remote end of the
+ *         connection.
+ */
+const struct sol_network_link_addr *sol_bt_conn_get_addr(
+    const struct sol_bt_conn *conn);
+
+/**
+ * @brief Attempts to establish a connection with a remote device.
+ *
+ * @param addr The network link address of the remote device,
+ *        @see sol_bt_start_scan().
+ * @param on_connect The callback to be called when the connection
+ *        is established successfully.
+ * @param on_disconnect The callback to be called when the connection
+ *        is terminated, after established.
+ * @param on_error The callback to be called when the connection cannot
+ *        be established.
+ * @param user_data User data to be passed to the connection callbacks.
+ *
+ * @return a reference to a connection, so the connection attempt can be cancelled,
+ *         @see sol_bt_disconnect().
+ */
+struct sol_bt_conn *sol_bt_connect(const struct sol_network_link_addr *addr,
+    bool (*on_connect)(void *user_data, struct sol_bt_conn *conn),
+    void (*on_disconnect)(void *user_data, struct sol_bt_conn *conn),
+    void (*on_error)(void *user_data, int error),
+    const void *user_data);
+
+/**
+ * @brief Terminates a connection, or connection attempt.
+ *
+ * In case the connection is not already established, the connection attempt
+ * will be cancelled after this is called.
+ *
+ * @param conn The reference to a connection.
+ *
+ * @return 0 on success, -errno otherwise.
+ */
+int sol_bt_disconnect(struct sol_bt_conn *conn);
+
+/**
+ * @brief Represents a Bluetooth usage session
+ *
+ * Because Bluetooth usage may increase the power comsuption, there's
+ * a need to keep track of what is using Bluetooth in the system, and
+ * keep Bluetooth turned off if it's not used.
+ */
+struct sol_bt_session;
+
+/**
+ * @brief Enables the local Bluetooth controller.
+ *
+ * Before using any other functionality from this module, this function
+ * should be called.
+ *
+ * @param enabled Function to be called when the controller changes state.
+ * @param user_data User data to be provided to the enabled() callback.
+ *
+ * @return a reference to a session, used for returning the system to its
+ * previous state, @see sol_bt_disable().
+ */
+struct sol_bt_session *sol_bt_enable(
+    void (*enabled)(void *data, bool powered),
```

Comment:
I am confuse with the usage of `powered`.

When this function is called it tryies to power on the adapter, in case of fail it will return NULL and the callback `enabled` will neve be called. So, this should be used to monitor when the adapter is powered off ?